modules/core/client/core.client.module.js

```diff
@@ -1,6 +1,8 @@
-'use strict';
+(function (app) {
```

Comment:
Why not just send in window and use window.applicationConfiguration.registerModule since that is what you are doing with init.  To me it makes more sense.  If i just open the file i know what window is but have no idea what ApplicationConfiguration is.